Add release notes for the Insights Operator for 4.18 (#88584)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

Red{nbsp}Hat uses the collected runtime workload data to gain insights that can help you make investment decisions that will drive and optimize how you use your {product-title} containers.
For more information, see xref:../nodes/clusters/nodes-cluster-enabling-features.adoc#nodes-cluster-enabling[_Enabling features using feature gates_].

[id="ocp-release-notes-insights-operator-rapid-recommendations_{context}"]

> ==== Rapid Recommendations

Reviewer: ```suggestion
==== Rapid recommendations and data collection enhancements for Insights Operator
```

Reviewer: See comment on line 503

> In this release, enhancements have been made to the Rapid Recommendations mechanism for remotely configuring the rules that determine the data that the Insights Operator collects.

Reviewer: ```suggestion
In this release, the rapid recommendations feature is enhanced by becoming version-independent, and builds on the existing conditional data gathering mechanism. The Insights Operator connects to a secure remote endpoint service running on [/console.redhat.com](link:https://console.redhat.com[console.redhat.com]) to retrieve definitions that contain the rules for determining which container log messages are filtered and collected by Red{nbsp}Hat.
```

Author: This is technically inaccurate. The feature was version-independent in the previous release but now further under-cover enhancements have been made for performance and scaling etc...
As this feature is still quite new and key to the Insights Operator evolution, the PM and engineering team want us to reiterate what we had for 4.17 as they have significantly enhanced the feature under the covers.

Reviewer: No worries, my general suggestion is based on minimalism. I was trying to find ways to trim some of the text. I leave technical accuracy to you.

> The Rapid Recommendations feature is version-independent, and builds on the existing conditional data gathering mechanism. 

Reviewer: ```suggestion

```

> The Insights Operator connects to a secure remote endpoint service running on /console.redhat.com to retrieve definitions that contain the rules for determining which container log messages are filtered and collected by Red{nbsp}Hat.

Reviewer: ```suggestion

```

> The conditional data-gathering definitions get configured through an attribute named `conditionalGathererEndpoint` in the link:https://github.com/openshift/insights-operator/blob/master/config/pod.yaml[`pod.yml`] configuration file.

Reviewer: We don't like to link to YAML files like this in our docs. I _think_ as an alternative you could maybe do a snippet like this:
[source,yaml]
# ...
<that field>
# ...

Author: The pod.yml used by Red Hat to collect customer data is dynamic and might change between OCP releases. If we add the yml content to the release notes, the docs could become obsolete soon after GA. We linked to it in the previous release also and this was approved by OCP docs.

Reviewer 2: As long as it's somewhere that requires Red Hat QE review for all changes, it should be ok. General GitHub links are not allowed because of the supportability concerns that unverified changes introduce, and this project isn't one that we know is always verified.

[source,bash]
----
conditionalGathererEndpoint: https://console.redhat.com/api/gathering/v2/%s/gathering_rules
----

> [NOTE]

Reviewer: You could rework that first sentence to include this note if you wanted. Something like, `Previously, the rules for the rules for determining the data that the Insights Operator collects were hard-coded and tied to the corresponding {product-title} version, with this update the rapid recommendations feature is enhanced by becoming version-independent. In addition, Insights Operator The Insights Operator now gathers more data to detect the following scenarios, which other applications can use to generate remedial recommendations to proactively manage your {product-title} deployments:

// Engineering reference: CCXDEV-14148
* Collects resources from the `nmstate.io/v1` API group.

// Engineering reference: CCXDEV-14521
*  Collects data from `clusterrole.rbac.authorization.k8s.io/v1` instances.

Author: Thank you for the suggestion. This isn't technically correct as per previous comments.

====
In earlier iterations, the rules for determining the data that the Insights Operator collects were hard-coded and tied to the corresponding {product-title} version.
====

The preconfigured endpoint URL now provides a placeholder (`%s`) for defining a target version of {product-title}.

[id="ocp-release-notes-insights-operator-more_data_collected_and_recommendations_added_{context}"]

> ==== More data collected and recommendations added

Reviewer: This is a pretty vague and general heading. I'd be more specific or really you could collapse this into the enhancements piece above.

Author: This heading is consistent with previous release notes and in other places for Insights prod docs:

https://docs.openshift.com/container-platform/4.17/release_notes/ocp-4-17-release-notes.html#more-data-collected-and-recommendations-added

We have a large epic coming for how we document the data that Insights collects in 4.19. Will add this feedback to that epic. Thanks.

|Supported
|====

> [id="ocp-release-notes-insights-operator-enhancements_{context}"]

Reviewer 3: Looks like this anchor was deleted inadvertently? Should add this back before merge

Red{nbsp}Hat uses the collected runtime workload data to gain insights that can help you make investment decisions that will drive and optimize how you use your {product-title} containers.
For more information, see xref:../nodes/clusters/nodes-cluster-enabling-features.adoc#nodes-cluster-enabling[_Enabling features using feature gates_].

[id="ocp-release-notes-insights-operator-rapid-recommendations_{context}"]
==== Rapid Recommendations

In this release, enhancements have been made to the Rapid Recommendations mechanism for remotely configuring the rules that determine the data that the Insights Operator collects.

The Rapid Recommendations feature is version-independent, and builds on the existing conditional data gathering mechanism. 

> The Insights Operator connects to a secure remote endpoint service running on /console.redhat.com to retrieve definitions that contain the rules for determining which container log messages are filtered and collected by Red{nbsp}Hat.

Reviewer 3: I assume this is to escape URL parsing - if so, you want the other slash :)
```suggestion
The Insights Operator connects to a secure remote endpoint service running on \console.redhat.com to retrieve definitions that contain the rules for determining which container log messages are filtered and collected by Red{nbsp}Hat.
```

[id="ocp-release-note-insights-operator-bug-fixes_{context}"]
==== Insights Operator

> Previously, during entitled builds on a Red{nbsp}Hat {product-title} cluster running on IBM-Z hardware, repositories were not enabled. This issue has been resolved. You can now enable repositories during entitled builds on a Red{nbsp}Hat {product-title}  cluster running on IBM-Z hardware. (OCPBUG-32233)

Reviewer: ```suggestion
Previously, during entitled builds on a  Red{nbsp} {product-title} cluster running on {ibm-z-title} hardware, repositories were not enabled. This issue has been resolved. and you can now enable repositories (link:https://issues.redhat.com/browse/OCPBUGS-32233[*OCPBUGS-32233*])
```

Author: There are punctuation issues and spacing typos in your suggestion above. 

Adding the attributes in another commit... Thanks.

Reviewer: Sorry, about that. I take it that you got the general suggestion that you were missing attributes and didn't format the bug properly.
